File: src/main/java/com/teamwizardry/libarianlib/client/gui/IGuiDrawable.java
package com.teamwizardry.libarianlib.client.gui;

import com.teamwizardry.libarianlib.math.Vec2;

/**
 * An object that can be drawn to a gui
 * @author Pierce Corcoran
 */
public interface IGuiDrawable {

	/**
	 * Draw this object to the screen.
	 * @param mousePos
	 * @param partialTicks
	 */
	void draw(Vec2 mousePos, float partialTicks);
	
}

File: src/main/java/com/teamwizardry/librarianlib/features/gui/IGuiDrawable.kt
package com.teamwizardry.librarianlib.features.gui

import com.teamwizardry.librarianlib.features.math.Vec2d
import net.minecraftforge.fml.relauncher.Side
import net.minecraftforge.fml.relauncher.SideOnly

/**
 * An object that can be drawn to a bookcomponents
 * @author Pierce Corcoran
 */
interface IGuiDrawable {

    /**
     * Draw this object to the screen.
     * @param mousePos
     * *
     * @param partialTicks
     */
    @SideOnly(Side.CLIENT)
    fun draw(mousePos: Vec2d, partialTicks: Float)

}
